frontend/js/src/utils/utils.tsx

```diff
@@ -465,8 +466,15 @@ const getPageProps = (): {
       critiquebrainz,
       sentry_traces_sample_rate,
       sentry_dsn,
+      user_preferences,
     } = globalReactProps;
 
+    if ("connection" in navigator) {
+      if (navigator.connection?.saveData === true) {
+        user_preferences!.saveDate = true;
```

Comment:
I don't like the use of `!` here, it's usually a bad code smell to force typescript types like that.
Can it be avoided? In this case, what was the reason for using this that user_preferences is possibly undefined (looking at GlobalAppProps type)?

It should be possible to instead ensure that it is an existing object.
Maybe something like:
```
user_preferences = Object.assign({}, user_preferences, {saveDate: true})
```
This ensures user_preferences is an existing empty array at least, and concatenate it with the user_preferences (if any) present in the page props, and also with an object that sets `saveData`.

This will require you to change the `const {` on line 461 to a `let {`, otherwise typescript will complain that you `cannot assign to user_preferences because it is a constant`

Reply:
Extracted the user property as let and the other's are left as const since eslint was throwing an error otherwise.
  `userPreferences` object is now reassigned as `user_preferences = { ...user_preferences, saveData: true };`